overlord/snapshotstate/snapshotstate.go

```diff
@@ -197,6 +202,30 @@ func Restore(st *state.State, setID uint64, snapNames []string, users []string)
 
 	ts = state.NewTaskSet()
 
+	for i, snapID := range snapIDs {
+		if snapID == "" {
+			// snapshotted snap was sideloaded, never mind
+			continue
+		}
+		snapst, ok := all[snapsFound[i]]
+		if !ok {
+			// snap not installed
+			continue
+		}
+		sideInfo := snapst.CurrentSideInfo()
+		if sideInfo == nil {
+			// ... shouldn't happen (but also not installed)
+			continue
+		}
+		if sideInfo.SnapID == "" {
+			// current snap is sideloaded, never mind then
+			continue
+		}
+		if sideInfo.SnapID != snapID {
+			return nil, nil, fmt.Errorf("cannot restore snapshot over id change: %.7s… → %.7s…", snapID, sideInfo.SnapID)
```

Comment:
Will the -> arrow render OK on simple terminals? Perhaps call a helper to translate it to "->"

Reply:
It works on the linux console with default settings, which was the good-enough-for-now pass mark we've used before.